include/chainbase/undo_index.hpp

```diff
@@ -0,0 +1,843 @@
+#pragma once
+
+#include <boost/multi_index_container_fwd.hpp>
+#include <boost/intrusive/set.hpp>
+#include <boost/intrusive/avltree.hpp>
+#include <boost/intrusive/slist.hpp>
+#include <boost/container/deque.hpp>
+#include <boost/throw_exception.hpp>
+#include <boost/mpl/fold.hpp>
+#include <boost/mp11/list.hpp>
+#include <boost/mp11/algorithm.hpp>
+#include <boost/iterator/transform_iterator.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/core/demangle.hpp>
+#include <boost/interprocess/interprocess_fwd.hpp>
+#include <memory>
+#include <type_traits>
+#include <sstream>
+
+namespace chainbase {
+
+   template<typename F>
+   struct scope_exit {
+    public:
+      scope_exit(F&& f) : _f(f) {}
```

Comment:
It doesn't actually matter as it's only used with capture-by-reference lambdas.

Reply:
Also, that would require adding a deduction guide, and serves no actual purpose here.  This scope_exit is only here to avoid a dependency on fc and doesn't need to be generalized beyond what chainbase itself uses.